tests/framework/nodeplacement.go

```diff
@@ -22,18 +22,29 @@ var (
 // The values chosen are valid, but the pod will likely not be schedulable.
 func (f *Framework) TestNodePlacementValues() sdkapi.NodePlacement {
 	nodes, _ := f.K8sClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
-	affinityTestValue = &v1.Affinity{
-		NodeAffinity: &v1.NodeAffinity{
-			RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
-				NodeSelectorTerms: []v1.NodeSelectorTerm{
-					{
-						MatchExpressions: []v1.NodeSelectorRequirement{
-							{Key: "kubernetes.io/hostname", Operator: v1.NodeSelectorOpIn, Values: []string{nodes.Items[0].Name}},
+
+	nodeIdx := -1
```

Comment:
Since we are really just interested in the node name, why not just store it? that way you don't have to keep track of the index, and you can initialize it with something like this:
```go
nodeName := nodes.Items[0].Name
for _, node := range nodes.Items {
  if _, hasLabel := node.Labels["node-role.kubernetes.io/worker"]; hasLabel {
    nodeName = node.Name
    break
  }
}
```
and then you can use the nodeName in the selector.

Reply:
Or if you don't want to initalize it, leave it blank and check if it is set after the loop

Reply:
sure, will fix.